lib/browser_crawler/engine.rb

```diff
@@ -121,15 +123,32 @@ def full_url(uri)
       end
     end
 
+    def skip_visit?(visited_page_link, uri)
+      visited_pages.include?(visited_page_link) ||
+        uri.host.nil? ||
+        uri.scheme.nil?
+    end
+
     def crawl(url:, only_path:)
-      return "Skipped external #{url}." unless internal_url?(url)
-      return 'Limit reached' if limit_reached?
+      puts("Skipped external #{url}.") && return unless internal_url?(url)
```

Comment:
is it necessary to log skipped urls?

Reply:
@Teckden  I think yes because we have just started to use this tool recently and don't know what kind of situation we can meet.